swaybar/render.c

```diff
@@ -154,15 +154,14 @@ static uint32_t render_status_block(cairo_t *cairo,
 		block_width += block->border_right + margin;
 	}
 
-	int sep_width;
+	int sep_width, sep_height;
 	if (!edge) {
 		if (config->sep_symbol) {
-			int _height;
-			get_text_size(cairo, config->font, &sep_width, &_height,
+			get_text_size(cairo, config->font, &sep_width, &sep_height,
 					output->scale, false, "%s", config->sep_symbol);
-			uint32_t _ideal_height = _height + ws_vertical_padding * 2;
-			if ((uint32_t)_height < _ideal_height / output->scale) {
-				return _height / output->scale;
+			uint32_t _ideal_height = sep_height + ws_vertical_padding * 2;
+			if (_ideal_height * output->scale > height) {
```

Comment:
Can you rename `_ideal_height` to `_ideal_surface_height` (so that it's explicit that this value is in surface-local coordinates, not in buffer-local coordinates)?

Also we should always compare sizes in surface-local coordinates to prevent rounding issues when the size is odd (see #1780). Se we should replace the comparison with `_ideal_surface_height > surface_height`.